src/riak_kv_qry_worker.erl

```diff
@@ -447,11 +429,11 @@ subqueries_done(QId, #state{qid          = QId,
                                     [[riak_pb_ts_codec:ldbvalue()]]}.
 prepare_final_results(#state{qbuf_ref = undefined,
                              result = IndexedChunks,
-                             qry = ?SQL_SELECT{'SELECT' = #riak_sel_clause_v1{calc_type = rows} = Select}}) ->
+                             qry = ?SQL_SELECT{'SELECT' = #riak_sel_clause_v1{calc_type = rows} = Select} = Query}) ->
     %% sort by index, to reassemble according to coverage plan
     {_, R2} = lists:unzip(lists:sort(IndexedChunks)),
```

Comment:
How does the naive sorting performed here play with the DESC qualifier in the PK definition? Consider a query with WHERE range that fetches 20 records, and consider it has a LIMIT value of 10. If the interesting column has been declared DESC in the DDL, then is the expectation to retrieve the top ten records, or the bottom ten? It appears it will be the latter.

Incidentally, this is also the behaviour in https://github.com/basho/riak_kv/pull/1587, but it has the resulting order specified separately, explicitly or implied via ASC/DESC qualifier in the ORDER BY clause. I am wondering if this should be clarified in the official docs (https://github.com/basho/private_basho_docs/pull/299).

Still, it would probably be great if,_ in absence of ORDER BY_, the DESC in the DDL did take effect. It would enable retrieving the actual last records, and thus easily emulate LAST.